### PR TITLE
Package glslang and add demos use to media-libs/vulkanloader

### PR DESCRIPTION
**Pull request by sarnex**
Hi all,

These changes  package glslang, which has no deps, which is required to build the vulkan demos in vulkan-loader.

Then, I add a USE to vulkan-loader for this.

The glslang releases are really old and vulkan-loader with +demos will not compile with even the latest one. So, I added a snapshot version and a live version.

Partially closes: https://bugs.gentoo.org/619124

I will work on the "layers" USE after this, where I have to package the SPIRV-Tools also.

@mattst88 

Please review.

Thanks,
Sarnex

### CODE REVIEW

**Comment by sarnex** [comment]
If you saw any of the previous versions, I decided to snapshot glslang also, so non-vulkan-loader-9999 users could use the USE.

Thanks

**Comment by sarnex** [comment]
@mattst88 Hey Matt,

Have you had a chance to take a look at these changes? If there something I should work on before pulling it, please let me know.

Thanks.
Sarnex

File: dev-util/glslang/glslang-3.0_p20171125.ebuild [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit cmake-multilib cmake-utils
+
+if [[ ${PV} == "9999" ]] ; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/KhronosGroup/glslang.git"
+	SRC_URI=""
```
Comment (SoapGentoo) [inline dev-util/glslang/glslang-3.0_p20171125.ebuild]: remove

File: dev-util/glslang/glslang-3.0_p20171125.ebuild [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+else
+	SNAPSHOT_COMMIT="792a94809b68d7403ec7f73f59b318bed9a4c020"
+	SRC_URI="https://github.com/KhronosGroup/${PN}/archive/${SNAPSHOT_COMMIT}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64"
+	S="${WORKDIR}/${PN}-${SNAPSHOT_COMMIT}"
```
Comment (SoapGentoo) [inline dev-util/glslang/glslang-3.0_p20171125.ebuild]: consider `vcs-snapshot` for this

File: dev-util/glslang/glslang-3.0_p20171125.ebuild [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+DESCRIPTION="Khronos reference front-end for GLSL and ESSL, and sample SPIR-V generator"
+HOMEPAGE="https://www.khronos.org/opengles/sdk/tools/Reference-Compiler/"
+
+LICENSE="BSD"
+SLOT="0"
+
+src_install() {
+	cmake-utils_src_install
+}
```
Comment (SoapGentoo) [inline dev-util/glslang/glslang-3.0_p20171125.ebuild]: why are you overriding this?

File: media-libs/vulkan-loader/vulkan-loader-9999.ebuild [hunk 4]
```diff
@@ -32,7 +33,7 @@ multilib_src_configure() {
 		-DCMAKE_SKIP_RPATH=True
 		-DBUILD_TESTS=False
```
Comment (SoapGentoo) [inline media-libs/vulkan-loader/vulkan-loader-9999.ebuild]: what about when you actually want to run tests?

File: dev-util/glslang/glslang-9999.ebuild [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+fi
+
+DESCRIPTION="Khronos reference front-end for GLSL and ESSL, and sample SPIR-V generator"
+HOMEPAGE="https://www.khronos.org/opengles/sdk/tools/Reference-Compiler/"
+
+LICENSE="BSD"
+SLOT="0"
+
+src_install() {
+	cmake-utils_src_install
```
Comment (SoapGentoo) [inline dev-util/glslang/glslang-9999.ebuild]: same here obviously

File: dev-util/glslang/glslang-3.0_p20171125.ebuild (hunk 1 above)
Comment (sarnex) [reply dev-util/glslang/glslang-3.0_p20171125.ebuild]: fixed

File: dev-util/glslang/glslang-3.0_p20171125.ebuild (hunk 2 above)
Comment (sarnex) [reply dev-util/glslang/glslang-3.0_p20171125.ebuild]: good idea thanks

File: dev-util/glslang/glslang-3.0_p20171125.ebuild (hunk 3 above)
Comment (sarnex) [reply dev-util/glslang/glslang-3.0_p20171125.ebuild]: because im an idiot

File: dev-util/glslang/glslang-9999.ebuild (hunk 5 above)
Comment (sarnex) [reply dev-util/glslang/glslang-9999.ebuild]: same here

File: media-libs/vulkan-loader/vulkan-loader-9999.ebuild (hunk 4 above)
Comment (sarnex) [reply media-libs/vulkan-loader/vulkan-loader-9999.ebuild]: i didnt write that part, but i added the use

**Comment by sarnex** [comment]
@SoapGentoo All concerns address, let me know if you have more feedback. Thanks for the review.

**Comment by mattst88** [comment]
Sorry, no, I haven't had a chance to review, and I'm kind of swamped.

**Comment by sarnex** [comment]
@SoapGentoo any comments on my latest changes?

**Comment by sarnex** [comment]
Are either of you more free recently? @SoapGentoo @mattst88 

Thank you.

**Comment by mattst88** [comment]
The glslang commit looks good. I'd change the commit SHA to the latest tag (6.2.2596) which is new enough for vulkan-loader-1.1.70.0 (which is what we should bump to).

However, I cannot make vulkan-loader compile: I get

`make[2]: *** No rule to make target '/var/tmp/portage/media-libs/vulkan-loader-9999/work/vulkan-loader-9999/external/glslang/External/spirv-tools/.git/index', needed by 'spirv_tools_commit_id.h'.  Stop.`

Any ideas?

**Comment by sarnex** [comment]
@mattst88 
Hi, thank you for the feedback. I have applied the glslang changes as you suggested.

The error was related to the external headers not being checked out. I fixed this by calling the relevant script. Please let me know if there is a better way to do this.

Please review.

Thanks,
Sarnex

**Comment by mattst88** [comment]
Unfortunately it looks like the script is actually downloading things from github -- which isn't allowed. Also, it's downloading glslang rather than using the one we have installed. Hmm.

**Comment by sarnex** [comment]
Yeah, that makes sense. After some investigation, I found that firstly this only applies to 9999, 1.0.61 builds fine without external sources.

Second, it looks like it doesn't need any of the external sources to compile besides the make rule causing the failure. I have just added a patch to remove the make rule. I have done some testing and everything seems to work.

Sorry if these changes are spaghetti, I haven't worked with this scenario before.

Please review,
Sarnex

**Comment by mattst88** [comment]
Thanks. I've pushed this with a couple of changes.

I dropped the changes to 1.0.61.1. People get cranky about changing dependencies without a revision bump. I also had to drop the test USE flag. With FEATURES=test enabled, it failed to build. Would be nice to sort that out.

1.0.68.0 and 1.1.70.0 are out, so we should get them into the tree. Unfortunately they don't compile for me, and the error message isn't informative. Hopefully you can take a look and figure it out.

Thanks for your contributions and also your patience. :)

**Comment by sarnex** [comment]
Thanks for the feedback. I will investigate the test and new version failures, package them, and then work on the layers use.

Thanks again :)

**Pull request closed without merging**